acceptance/tests/resource/file/source_attribute.rb

```diff
@@ -11,7 +11,23 @@
   @target_dir_on_windows  = 'C:/windows/temp/source_attr_test_dir'
   @target_dir_on_nix      = '/tmp/source_attr_test_dir'
 
-  checksums = [nil, 'md5', 'md5lite', 'sha256', 'sha256lite', 'ctime', 'mtime']
+  # In case any of the hosts happens to be fips enabled we limit to the lowest
+  # common denominator.
+  checksums_fips = [nil, 'sha256', 'sha256lite', 'ctime', 'mtime']
+  checksums_no_fips = [nil, 'md5', 'md5lite', 'sha256', 'sha256lite', 'ctime', 'mtime']
+ 
+  fips_host_present = 0
+  hosts.each do |host|
+    if (on(host, facter("find fips_enabled")).stdout =~ /true/)
+      fips_host_present = 1
+    end
+  end
```

Comment:
Since we want to know if there is at least one fips host present, we can stop as soon as we find the first one. A ruby way of doing that is to write:

```ruby
fips_host_present = hosts.any? { |host| on(host, facter("fips_enabled")).stdout =~ /true/ }
```

Reply:
yeah am still learning all such compact ways of expressing things. thanks.